Updates to creation of new cycle/report

- [x] Fetch latest reports on creation of new one so refresh is not necessary and new report is shown on open report cycle (FE) admin action
- [x] Fix error on new report creation where no pdf file generated
- [x] Fix so that latest report participating and pending report tabs are populated with latest data on batch approval
- [x] Fix so that active projects are converted to update requested on opening a new cycle (or else the latest pr is approved automatically)
- [x] Fix so that student reports are also set to update requested (or else all are approved on update requested/batchapprove)
- [x] Fix so that deleting report info deletes both the progress reports and student reports (seems student reports remained - causing this error as well:   ...service-spms-xEZ16XFH-py3.12\Lib\site-packages\django\db\models\query.py", line 652, in get
    raise self.model.MultipleObjectsReturned(
documents.models.StudentReport.MultipleObjectsReturned: get() returned more than one StudentReport -- it returned 2!
22-04-2024 @ 14:52:54 [ERROR] "GET /api/v1/documents/studentreports/528/2023 HTTP/1.1" 500 136973)
- [x] Remove extra pdf gens (_Annual_Report_12_Ec77MH1 instead of _Annual_Report_12 being replaced), and ensure the old version is deleted before being replaced with a new one with same name

--- a/src/components/Modals/NewCycleModal.tsx
+++ b/src/components/Modals/NewCycleModal.tsx
@@ -20,7 +20,7 @@ import {
   useDisclosure,
   useToast,
 } from "@chakra-ui/react";
-import { useMutation } from "@tanstack/react-query";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
 import { useEffect, useRef, useState } from "react";
 import {
   INewCycle,
@@ -52,7 +52,7 @@ export const NewCycleModal = ({ isOpen, onClose }: IModalProps) => {
     onClose();
   };
 
-  const [shouldIncludeUpdate, setShouldIncludeUpdate] = useState(false);
+  const [shouldIncludeUpdate, setShouldIncludeUpdate] = useState(true);
   const [shouldSendEmails, setShouldSendEmails] = useState(false);
 
   // Toast
@@ -61,6 +61,7 @@ export const NewCycleModal = ({ isOpen, onClose }: IModalProps) => {
   const addToast = (data) => {
     toastIdRef.current = toast(data);
   };
+  const queryClient = useQueryClient();
 
   const newCycleMutation = useMutation<
     MutationSuccess,
@@ -90,6 +91,11 @@ export const NewCycleModal = ({ isOpen, onClose }: IModalProps) => {
           isClosable: true,
         });
       }
+      queryClient.invalidateQueries({
+        queryKey: ["latestUnapprovedProgressReports"],
+      });
+      queryClient.invalidateQueries({ queryKey: ["latestProgressReports"] });
+      queryClient.invalidateQueries({ queryKey: ["latestStudentReports"] });
       //  Close the modal
       if (onClose) {
         onClose();

--- a/src/components/Modals/RTEModals/ApproveProgressReportModal.tsx
+++ b/src/components/Modals/RTEModals/ApproveProgressReportModal.tsx
@@ -1,0 +1,312 @@
+import { ExtractedHTMLTitle } from "@/components/ExtractedHTMLTitle";
+import { IProgressReportDisplayData } from "@/components/RichTextEditor/Editors/ARProgressReportHandler";
+import { IStudentReportDisplayData } from "@/components/RichTextEditor/Editors/ARStudentReportHandler";
+import { IApproveProgressReport, approveProgressReport } from "@/lib/api";
+import {
+  Box,
+  Button,
+  Flex,
+  FormControl,
+  Grid,
+  Input,
+  InputGroup,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  ToastId,
+  useColorMode,
+  useToast,
+} from "@chakra-ui/react";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { useRef } from "react";
+import { useForm } from "react-hook-form";
+
+interface Props {
+  report: IProgressReportDisplayData | IStudentReportDisplayData;
+  isActive: boolean;
+  isOpen: boolean;
+  onClose: () => void;
+  isAnimating?: boolean;
+  setIsAnimating?: (state: boolean) => void;
+}
+
+export const ApproveProgressReportModal = ({
+  report,
+  isActive,
+  setIsAnimating,
+  isOpen,
+  onClose,
+}: Props) => {
+  const { colorMode } = useColorMode();
+  const queryClient = useQueryClient();
+  const { register, handleSubmit, reset } = useForm<IApproveProgressReport>();
+  const toast = useToast();
+  const toastIdRef = useRef<ToastId>();
+  const addToast = (data) => {
+    toastIdRef.current = toast(data);
+  };
+
+  const approveProgressReportMutation = useMutation({
+    mutationFn: approveProgressReport,
+    onMutate: () => {
+      addToast({
+        status: "loading",
+        title: isActive
+          ? `Setting Progress Report to Pending`
+          : `Approving Progress Report`,
+        position: "top-right",
+      });
+    },
+    onSuccess: () => {
+      if (setIsAnimating) {
+        setIsAnimating(true);
+      }
+
+      if (toastIdRef.current) {
+        toast.update(toastIdRef.current, {
+          title: "Success",
+          description: isActive ? `Set to Pending` : `Approved`,
+          status: "success",
+          position: "top-right",
+          duration: 3000,
+          isClosable: true,
+        });
+      }
+      reset();
+      queryClient.invalidateQueries({
+        queryKey: ["latestUnapprovedProgressReports"],
+      });
+      queryClient.invalidateQueries({ queryKey: ["latestProgressReports"] });
+      queryClient.invalidateQueries({ queryKey: ["latestStudentReports"] });
+      onClose();
+
+      setTimeout(() => {
+        if (setIsAnimating) {
+          setIsAnimating(false);
+        }
+      }, 350);
+    },
+    onError: (error) => {
+      if (toastIdRef.current) {
+        toast.update(toastIdRef.current, {
+          title: isActive ? `Could Not Set to Pending` : `Could Not Approve`,
+          description: `${error}`,
+          status: "error",
+          position: "top-right",
+          duration: 3000,
+          isClosable: true,
+        });
+      }
+    },
+  });
+
+  const onBeginApprovalMutation = (formData: IApproveProgressReport) => {
+    approveProgressReportMutation.mutate(formData);
+  };
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} size={"lg"}>
+      <ModalOverlay />
+      <ModalContent
+        color={colorMode === "light" ? "black" : "white"}
+        bg={colorMode === "light" ? "white" : "gray.800"}
+      >
+        <ModalHeader>
+          {isActive ? "Return" : "Approve"} Progress Report?
+        </ModalHeader>
+        <ModalCloseButton />
+
+        <ModalBody
+          as="form"
+          id="progressreportapprove-form"
+          onSubmit={handleSubmit(onBeginApprovalMutation)}
+        >
+          <FormControl pb={6}>
+            <InputGroup>
+              <Input
+                {...register("kind", { required: true })}
+                type="hidden"
+                defaultValue={report.document.kind}
+              />
+              <Input
+                {...register("reportPk", { required: true })}
+                type="hidden"
+                defaultValue={report.pk}
+              />
+              <Input
+                {...register("documentPk", { required: true })}
+                type="hidden"
+                defaultValue={report.document.pk}
+              />
+              <Input
+                {...register("isActive", { required: true })}
+                type="hidden"
+                defaultValue={isActive ? 1 : 0}
+              />
+            </InputGroup>
+          </FormControl>
+
+          <Box>
+            <Text fontWeight={"bold"}>
+              {isActive
+                ? "Set this progress report to pending"
+                : "Provide final sign off for this progress report"}
+              ?
+            </Text>
+            <Text>
+              {isActive
+                ? "Note: this will set the project back to update requested and set the document's directorate approval to Required"
+                : "Note: If not already approved by project lead and business area lead, the report will be fast-tracked and provided with all approvals"}
+            </Text>
+            <Grid gridTemplateColumns={"repeat(1, 1fr)"} mt={2}>
+              <Flex mb={4} mt={2}>
+                <Box>
+                  <ExtractedHTMLTitle
+                    htmlContent={report?.document?.project?.title}
+                    color={"green.400"}
+                  />
+                </Box>
+              </Flex>
+              <Box
+                justifyContent={"center"}
+                textAlign={"center"}
+                display={"flex"}
+                ml={"-10%"}
+              >
+                <Flex w={"50%"}>
+                  {" "}
+                  <Box
+                    flex={1}
+                    justifyContent={"flex-end"}
+                    textAlign={"end"}
+                    mr={4}
+                  >
+                    <Text fontWeight={"bold"}>Kind:</Text>
+                  </Box>
+                  <Box minW={"120px"} textAlign={"start"}>
+                    <Text>{report?.document?.kind}</Text>
+                  </Box>
+                </Flex>
+              </Box>
+
+              <Box
+                justifyContent={"center"}
+                textAlign={"center"}
+                display={"flex"}
+                ml={"-10%"}
+              >
+                <Flex w={"50%"}>
+                  {" "}
+                  <Box
+                    flex={1}
+                    justifyContent={"flex-end"}
+                    textAlign={"end"}
+                    mr={4}
+                  >
+                    <Text fontWeight={"bold"}>Year:</Text>
+                  </Box>
+                  <Box minW={"120px"} textAlign={"start"}>
+                    <Text>{report?.year}</Text>
+                  </Box>
+                </Flex>
+              </Box>
+
+              <Box
+                justifyContent={"center"}
+                textAlign={"center"}
+                display={"flex"}
+                ml={"-10%"}
+              >
+                <Flex w={"50%"}>
+                  {" "}
+                  <Box
+                    flex={1}
+                    justifyContent={"flex-end"}
+                    textAlign={"end"}
+                    mr={4}
+                  >
+                    <Text fontWeight={"bold"}>Report PK:</Text>
+                  </Box>
+                  <Box minW={"120px"} textAlign={"start"}>
+                    <Text>{report?.pk}</Text>
+                  </Box>
+                </Flex>
+              </Box>
+
+              <Box
+                justifyContent={"center"}
+                textAlign={"center"}
+                display={"flex"}
+                ml={"-10%"}
+              >
+                <Flex w={"50%"}>
+                  {" "}
+                  <Box
+                    flex={1}
+                    justifyContent={"flex-end"}
+                    textAlign={"end"}
+                    mr={4}
+                  >
+                    <Text fontWeight={"bold"}>Document PK:</Text>
+                  </Box>
+                  <Box minW={"120px"} textAlign={"start"}>
+                    <Text>{report?.document?.pk}</Text>
+                  </Box>
+                </Flex>
+              </Box>
+
+              <Box
+                justifyContent={"center"}
+                textAlign={"center"}
+                display={"flex"}
+                ml={"-10%"}
+              >
+                <Flex w={"50%"}>
+                  {" "}
+                  <Box
+                    flex={1}
+                    justifyContent={"flex-end"}
+                    textAlign={"end"}
+                    mr={4}
+                  >
+                    <Text fontWeight={"bold"}>Action:</Text>
+                  </Box>
+                  <Box minW={"120px"} textAlign={"start"}>
+                    <Text>{isActive ? "Set to Pending" : "Approve"}</Text>
+                  </Box>
+                </Flex>
+              </Box>
+            </Grid>
+          </Box>
+        </ModalBody>
+
+        <ModalFooter>
+          <Button
+            // variant="ghost"
+            mr={3}
+            onClick={onClose}
+          >
+            Cancel
+          </Button>
+          <Button
+            form="progressreportapprove-form"
+            type="submit"
+            isLoading={approveProgressReportMutation.isPending}
+            bg={colorMode === "dark" ? "green.500" : "green.400"}
+            color={"white"}
+            _hover={{
+              bg: colorMode === "dark" ? "green.400" : "green.300",
+            }}
+          >
+            {isActive ? "Set to Pending" : "Approve"}
+          </Button>
+        </ModalFooter>
+      </ModalContent>
+    </Modal>
+  );
+};

--- a/src/components/Pages/ProjectDetail/DocActions/ProjectDocumentPDFSection.tsx
+++ b/src/components/Pages/ProjectDetail/DocActions/ProjectDocumentPDFSection.tsx
@@ -309,7 +309,9 @@ export const ProjectDocumentPDFSection = ({
             <Box mr={2}>
               <BsStars />
             </Box>
-            Generate PDF
+            {data_document?.document?.pdf?.file
+              ? "Generate New"
+              : "Generate PDF"}
           </Button>
         </Flex>
       </Flex>

--- a/src/components/Pages/ProjectDetail/ProjectOverviewCard.tsx
+++ b/src/components/Pages/ProjectDetail/ProjectOverviewCard.tsx
@@ -655,7 +655,7 @@ export const ProjectOverviewCard = ({
                         )?.organisation[0]?.toUpperCase()}${(
                           details?.student as IStudentProjectDetails
                         )?.organisation.slice(1)}`
-                      : null}
+                      : "No organisation listed"}
                   </Text>
                 </Flex>
                 <Flex alignItems="center">
@@ -667,7 +667,7 @@ export const ProjectOverviewCard = ({
                       ? levelToString(
                           (details?.student as IStudentProjectDetails)?.level
                         )
-                      : null}
+                      : "No level selected"}
                   </Text>
                 </Flex>
               </Grid>

--- a/src/components/RichTextEditor/Editors/ARProgressReportEditor.tsx
+++ b/src/components/RichTextEditor/Editors/ARProgressReportEditor.tsx
@@ -93,7 +93,7 @@ export const ARProgressReportEditor = ({
 
   const { colorMode } = useColorMode();
 
-  const navigate = useNavigate();
+  // const navigate = useNavigate();
 
   return (
     <>
@@ -131,10 +131,7 @@ export const ARProgressReportEditor = ({
                   />
                 </Box>
                 <Box ml={4} flex={1}>
-                  <Box
-                    cursor={"pointer"}
-                    onClick={() => navigate(`/projects/${project?.pk}/student`)}
-                  >
+                  <Box>
                     <ExtractedHTMLTitle
                       htmlContent={project?.title}
                       color={"blue.500"}
@@ -142,6 +139,14 @@ export const ARProgressReportEditor = ({
                       fontSize={"17px"}
                       // fontSize={"xs"}
                       noOfLines={4}
+                      cursor={"pointer"}
+                      onClick={() =>
+                        { 
+                          const url = `/projects/${project?.pk}/progress`;
+                          window.open(url, '_blank');
+                          // navigate(url);
+                        }
+                      }
                     />
                   </Box>
 
@@ -165,10 +170,7 @@ export const ARProgressReportEditor = ({
             ) : (
               <>
                 <Box mr={4} flex={1}>
-                  <Box
-                    cursor={"pointer"}
-                    onClick={() => navigate(`/projects/${project?.pk}/student`)}
-                  >
+                  <Box>
                     <ExtractedHTMLTitle
                       htmlContent={project?.title}
                       color={"blue.500"}
@@ -176,6 +178,14 @@ export const ARProgressReportEditor = ({
                       fontSize={"17px"}
                       // fontSize={"xs"}
                       noOfLines={4}
+                      cursor={"pointer"}
+                      onClick={() =>
+                        { 
+                          const url = `/projects/${project?.pk}/progress`;
+                          window.open(url, '_blank');
+                          // navigate(url);
+                        }
+                      }
                     />
                   </Box>
 

--- a/src/components/RichTextEditor/Editors/PREditor.tsx
+++ b/src/components/RichTextEditor/Editors/PREditor.tsx
@@ -13,6 +13,7 @@ import {
   Text,
   ToastId,
   useColorMode,
+  useDisclosure,
   useToast,
 } from "@chakra-ui/react";
 import { $generateHtmlFromNodes } from "@lexical/html";
@@ -34,13 +35,15 @@ import { $getRoot } from "lexical";
 import { useEffect, useRef, useState } from "react";
 import { useForm } from "react-hook-form";
 import { AiFillEdit, AiFillEyeInvisible } from "react-icons/ai";
-import { FaSave } from "react-icons/fa";
+import { FaSave, FaUndo } from "react-icons/fa";
 import { useNavigate } from "react-router-dom";
 import { CustomPastePlugin } from "../Plugins/CustomPastePlugin";
 import ListMaxIndentLevelPlugin from "../Plugins/ListMaxIndentLevelPlugin";
 import { PrepopulateHTMLPlugin } from "../Plugins/PrepopulateHTMLPlugin";
 import { RevisedRichTextToolbar } from "../Toolbar/RevisedRichTextToolbar";
 import { IProgressReportDisplayData } from "./ARProgressReportHandler";
+import { ApproveProgressReportModal } from "@/components/Modals/RTEModals/ApproveProgressReportModal";
+import { TiTick } from "react-icons/ti";
 
 interface IPREditorProps {
   // isEditing: boolean;
@@ -156,7 +159,7 @@ export const PREditor = ({
   setContextDisplayData,
   // key
 }: IPREditorProps) => {
-  const navigate = useNavigate();
+  // const navigate = useNavigate();
   const { colorMode } = useColorMode();
 
   // const editorInstance = useLexicalComposerContext();
@@ -429,10 +432,24 @@ export const PREditor = ({
     uneditableInitialConfig,
   ]);
 
+
+  const {isOpen: isApproveProgressReportOpen, onOpen:onOpenApproveProgressReport, onClose: onCloseApproveProgressReport} = useDisclosure();
+  const [reportHovered, setReportHovered] = useState(false);
+  const [isActive, setIsActive] = useState(fullPRData?.document?.project?.status === "active");
+  useEffect(() => 
+    {setIsActive(fullPRData?.document?.project?.status === "active")}
+, [fullPRData]);
+
+
   return (
     <>
       {/* setIsEditing */}
-
+      <ApproveProgressReportModal 
+        isActive={isActive}
+        report={fullPRData} 
+        isOpen={isApproveProgressReportOpen} 
+        onClose={onCloseApproveProgressReport}      
+    />
       <Box
         mx={4}
         pos={"relative"}
@@ -443,7 +460,47 @@ export const PREditor = ({
           "0 8px 24px rgba(0, 0, 0, 0.1), 0 2px 8px rgba(0, 0, 0, 0.1)"
         }
         bg={colorMode === "light" ? "whiteAlpha.600" : "blackAlpha.500"}
+        onMouseOver={() => setReportHovered(true)}
+        onMouseOut={() => setReportHovered(false)}
       >
+        {reportHovered ? 
+             <Box pos={"absolute"} right={4} top={4}>
+                    <Button
+                        ml={2}
+                        bg={
+                          isActive ?
+                          colorMode === "light" ? `orange.500` : `orange.600`
+                          :
+                          colorMode === "light" ? `green.500` : `green.600`}
+                        color={
+                          colorMode === "light"
+                            ? "whiteAlpha.900"
+                            : "whiteAlpha.800"
+                        }
+                        _hover={
+                          colorMode === "light"
+                            ? {
+                                bg: isActive ? `orange.600` : `green.600`,
+                                color: `white`,
+                              }
+                            : {
+                                bg: isActive ? `orange.500` : `green.500`,
+                                color: `white`,
+                              }
+                        }
+                        minW={"32px"}
+                        minH={"32px"}
+                        maxW={"32px"}
+                        maxH={"32px"}
+                        rounded={"full"}
+                        data-tip="Click to edit"
+                        onClick={onOpenApproveProgressReport}
+                      >
+                        <Icon as={isActive ? FaUndo : TiTick } />
+                      </Button>
+             </Box>
+        : null}
+   
         <Flex
           id={`topContent_${fullPRData?.document?.project?.pk}`}
           pt={6}
@@ -460,14 +517,7 @@ export const PREditor = ({
                 />
               </Box>
               <Box ml={4} flex={1}>
-                <Box
-                  cursor={"pointer"}
-                  onClick={() =>
-                    navigate(
-                      `/projects/${fullPRData?.document?.project?.pk}/progress`
-                    )
-                  }
-                >
+                <Box>
                   <ExtractedHTMLTitle
                     htmlContent={fullPRData?.document?.project?.title}
                     color={"blue.500"}
@@ -475,6 +525,14 @@ export const PREditor = ({
                     fontSize={"17px"}
                     // fontSize={"xs"}
                     noOfLines={4}
+                    cursor={"pointer"}
+                    onClick={() =>
+                      {
+                        const url = `/projects/${fullPRData?.document?.project?.pk}/progress`
+                        window.open(url, '_blank');
+                        // navigate(url);
+                      }
+                    }
                   />
                 </Box>
 
@@ -487,20 +545,21 @@ export const PREditor = ({
           ) : (
             <>
               <Box mr={4} flex={1}>
-                <Box
-                  cursor={"pointer"}
-                  onClick={() =>
-                    navigate(
-                      `/projects/${fullPRData?.document?.project?.pk}/progress`
-                    )
-                  }
-                >
+                <Box>
                   <ExtractedHTMLTitle
                     htmlContent={fullPRData?.document?.project?.title}
                     color={"blue.500"}
                     fontWeight={"bold"}
                     fontSize={"17px"}
                     // fontSize={"xs"}
+                    cursor={"pointer"}
+                    onClick={() =>
+                      {
+                        const url = `/projects/${fullPRData?.document?.project?.pk}/progress`
+                        window.open(url, '_blank');
+                        // navigate(url);
+                      }
+                    }
                     noOfLines={4}
                   />
                 </Box>

--- a/src/components/RichTextEditor/Editors/SREditor.tsx
+++ b/src/components/RichTextEditor/Editors/SREditor.tsx
@@ -10,6 +10,7 @@ import {
   Text,
   ToastId,
   useColorMode,
+  useDisclosure,
   useToast,
 } from "@chakra-ui/react";
 import { $generateHtmlFromNodes } from "@lexical/html";
@@ -31,13 +32,17 @@ import { $getRoot } from "lexical";
 import { useEffect, useRef, useState } from "react";
 import { useForm } from "react-hook-form";
 import { AiFillEdit, AiFillEyeInvisible } from "react-icons/ai";
-import { FaSave } from "react-icons/fa";
+import { FaSave, FaStamp, FaUndo } from "react-icons/fa";
 import { useNavigate } from "react-router-dom";
 import { CustomPastePlugin } from "../Plugins/CustomPastePlugin";
 import ListMaxIndentLevelPlugin from "../Plugins/ListMaxIndentLevelPlugin";
 import { PrepopulateHTMLPlugin } from "../Plugins/PrepopulateHTMLPlugin";
 import { RevisedRichTextToolbar } from "../Toolbar/RevisedRichTextToolbar";
 import { IStudentReportDisplayData } from "./ARStudentReportHandler";
+import { FcApproval } from "react-icons/fc";
+import { MdApproval } from "react-icons/md";
+import { TiTick } from "react-icons/ti";
+import { ApproveProgressReportModal } from "@/components/Modals/RTEModals/ApproveProgressReportModal";
 
 interface ISREditorProps {
   isEditing: boolean;
@@ -68,6 +73,7 @@ const SRProjDetails = ({ project, team_members }: ISRProjDetails) => {
   const students = getMembersByRole(team_members, "student");
   const academicSupervisors = getMembersByRole(team_members, "academicsuper");
   const scientists = getMembersByRole(team_members, "supervising");
+
 
   return (
     <Box py={3}>
@@ -110,7 +116,7 @@ export const SREditor = ({
   setDisplayData,
   // key
 }: ISREditorProps) => {
-  const navigate = useNavigate();
+  // const navigate = useNavigate();
   const { colorMode } = useColorMode();
 
   // const editorInstance = useLexicalComposerContext();
@@ -229,10 +235,22 @@ export const SREditor = ({
     setIsEditing(false);
   };
 
+  const {isOpen: isApproveProgressReportOpen, onOpen:onOpenApproveProgressReport, onClose: onCloseApproveProgressReport} = useDisclosure();
+
+  const [isActive, setIsActive] = useState(fullSRData?.document?.project?.status === "active");
+  useEffect(() => 
+    {setIsActive(fullSRData?.document?.project?.status === "active")}
+, [fullSRData]);
+
   return (
     <>
       {/* setIsEditing */}
-
+    <ApproveProgressReportModal 
+    isActive={isActive}
+    report={fullSRData} 
+    isOpen={isApproveProgressReportOpen} 
+    onClose={onCloseApproveProgressReport}      
+    />
       <Box
         mx={4}
         pos={"relative"}
@@ -263,12 +281,6 @@ export const SREditor = ({
               </Box>
               <Box ml={4} flex={1}>
                 <Box
-                  cursor={"pointer"}
-                  onClick={() =>
-                    navigate(
-                      `/projects/${fullSRData?.document?.project?.pk}/student`
-                    )
-                  }
                 >
                   <ExtractedHTMLTitle
                     htmlContent={fullSRData?.document?.project?.title}
@@ -277,6 +289,14 @@ export const SREditor = ({
                     fontSize={"17px"}
                     // fontSize={"xs"}
                     noOfLines={4}
+                    cursor={"pointer"}
+                    onClick={() =>
+                      { 
+                        const url = `/projects/${fullSRData?.document?.project?.pk}/student`;
+                        window.open(url, '_blank');
+                        // navigate(url);
+                      }
+                    }
                   />
                 </Box>
 
@@ -290,12 +310,6 @@ export const SREditor = ({
             <>
               <Box mr={4} flex={1}>
                 <Box
-                  cursor={"pointer"}
-                  onClick={() =>
-                    navigate(
-                      `/projects/${fullSRData?.document?.project?.pk}/student`
-                    )
-                  }
                 >
                   <ExtractedHTMLTitle
                     htmlContent={fullSRData?.document?.project?.title}
@@ -304,6 +318,14 @@ export const SREditor = ({
                     fontSize={"17px"}
                     // fontSize={"xs"}
                     noOfLines={4}
+                    cursor={"pointer"}
+                    onClick={() =>
+                    { 
+                      const url = `/projects/${fullSRData?.document?.project?.pk}/student`;
+                      window.open(url, '_blank');
+                      // navigate(url);
+                    }
+                    }
                   />
                 </Box>
 
@@ -439,7 +461,44 @@ export const SREditor = ({
                       </Flex>
                     </Box>
                   ) : isHovered ? (
-                    <Box pos={"absolute"} right={10} top={0}>
+                    <>
+                    <Box pos={"absolute"} right={8} top={0}>
+                    <Button
+                        ml={2}
+                        bg={
+                          isActive ?
+                          colorMode === "light" ? `orange.500` : `orange.600`
+                          :
+                          colorMode === "light" ? `green.500` : `green.600`}
+                        color={
+                          colorMode === "light"
+                            ? "whiteAlpha.900"
+                            : "whiteAlpha.800"
+                        }
+                        _hover={
+                          colorMode === "light"
+                            ? {
+                                bg: isActive ? `orange.600` : `green.600`,
+                                color: `white`,
+                              }
+                            : {
+                                bg: isActive ? `orange.500` : `green.500`,
+                                color: `white`,
+                              }
+                        }
+                        minW={"32px"}
+                        minH={"32px"}
+                        maxW={"32px"}
+                        maxH={"32px"}
+                        rounded={"full"}
+                        data-tip="Click to edit"
+                        onClick={onOpenApproveProgressReport}
+                      >
+                        <Icon as={isActive ? FaUndo : TiTick } />
+                      </Button>
+
+                    </Box>
+                    <Box pos={"absolute"} right={20} top={0}>
                       <Button
                         ml={2}
                         bg={colorMode === "light" ? `gray.500` : `gray.600`}
@@ -470,6 +529,8 @@ export const SREditor = ({
                         <Icon as={AiFillEdit} />
                       </Button>
                     </Box>
+                    </>
+                    
                   ) : null}
                 </Box>
 

--- a/src/lib/api.ts
+++ b/src/lib/api.ts
@@ -2,7 +2,7 @@ import { ICommentReaction } from "@/components/RichTextEditor/Editors/Sections/C
 import { QueryFunctionContext } from "@tanstack/react-query";
 import axios, { AxiosHeaders } from "axios";
 import Cookie from 'js-cookie';
-import { BusinessAreaImage, EditorSections, EditorSubsections, EditorType, IAddLocationForm, IAddress, IAffiliation, IApproveDocument, IBranch, IBusinessArea, IDepartmentalService, IDivision, IFeedback, IMergeAffiliation, IPersonalInformation, IProfile, IProjectMember, IQuickTask, IReport, IReportCreation, ISearchTerm, ISimpleLocationData, OrganisedLocationData, } from "../types";
+import { BusinessAreaImage, EditorSections, EditorSubsections, EditorType, IAddLocationForm, IAddress, IAffiliation, IApproveDocument, IBranch, IBusinessArea, IDepartmentalService, IDivision, IFeedback, IMergeAffiliation, IPersonalInformation, IProfile, IProgressReport, IProjectMember, IQuickTask, IReport, IReportCreation, ISearchTerm, ISimpleLocationData, OrganisedLocationData, } from "../types";
 import { IConceptPlanGenerationData } from "../types";
 
 
@@ -297,6 +297,27 @@ export const deactivateUserAdmin = async ({ userPk }: AdminSwitchVar) => {
 export const getMe = async () => {
     const res = instance.get(`users/me`).then(res => {
         // console.log(res.data)
+        return res.data
+    })
+    return res;
+}
+
+export interface IApproveProgressReport {
+    kind: "studentreport" | "progressreport";
+    isActive: number;
+    reportPk: number;
+    documentPk: number;
+}
+
+export const approveProgressReport = async ({isActive, kind, reportPk, documentPk}: IApproveProgressReport) => {
+    console.log({
+        kind,
+        reportPk,
+        documentPk, isActive,
+    })
+    const res = instance.post(`documents/actions/finalApproval`, {
+        kind, reportPk, documentPk, isActive: isActive.toString() === '1' ? true : false
+    }).then(res => {
         return res.data
     })
     return res;

